src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -196,6 +196,14 @@ protected virtual async Task ProduceTagsAsync(TaggerContext<TTag> context)
             foreach (var spanToTag in context.SpansToTag)
             {
                 context.CancellationToken.ThrowIfCancellationRequested();
+
+                // Let LSP handle producing tags in the cloud scenario
+                var workspaceContextService = spanToTag.Document?.Project.Solution.Workspace.Services.GetRequiredService<IWorkspaceContextService>();
```

Comment:
Based on some failing tests, apparently a DocumentSnapshotSpan's Document can be null, so had to add a check here.

Reply:
I don't think we want to disable folding ranges here - there's a bunch of taggers that derive from this, including line separators and brace highlighting which still need to run on the codespace client.